portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/SkinnableCharonPortal.java

```diff
@@ -1868,11 +1870,19 @@ public void includeLogin(PortalRenderContext rcontext, HttpServletRequest req,
 					loginUserDispName = Validator.escapeHtml(thisUser.getDisplayName());
 					loginUserFirstName = Validator.escapeHtml(thisUser.getFirstName());
 				}
+				
+				// check if current user is being impersonated (by become user/sutool)
+				String impersonatorDisplayId = getImpersonatorDisplayId();
+				if (!impersonatorDisplayId.isEmpty())
+				{
+					message = rloader.getString("sit_return") + " " + impersonatorDisplayId;
```

Comment:
Could you use a positional parameter here in the i18n string so that languages have flexibility in how they phrase it.